### PR TITLE
feat: 2020/11.

### PR DESCRIPTION
**Pull request by neverendingqs**

### CODE REVIEW

File: 2020/11/index.js [hunk 1]
```diff
@@ -1,3 +1,223 @@
...
+
+async function getInput() {
+  return input = (await fs.readFile(`${__dirname}/input.txt`))
+    .toString()
+    .trim()
+    .split('\n')
+    .map(row => row.split(''));
+}
+
+function p1NumAdjOccupiedSeats(input, row, column) {
```
Comment (faizanminhas) [inline 2020/11/index.js:11]: Oof, I hope you generated this code! I think my solution is very succinct and easier to understand [here](https://github.com/faizanminhas/advent-of-code-2020/pull/4/files#diff-2591cbae71189f9147ae87772a1bd929a92ac64f31d511b8700c99388786d03aR7)

File: 2020/11/index.js [hunk 2]
```diff
@@ -1,3 +1,223 @@
...
+
+  return numAdjOccupied;
+}
+function p2NumAdjOccupiedSeats(input, row, column) {
+  /*
+   * x x x      0 1 2
+   * x x x  =>  3 4 5
+   * x x x      6 7 8
+   */
+  const adj = {
```
Comment (faizanminhas) [inline 2020/11/index.js:50]: We had a pretty similar approach here

File: 2020/11/index.js [hunk 3]
```diff
@@ -1,3 +1,223 @@
...
+
+  function shouldContinue(key) {
+    if(!adj[key]) {
+      return false;
+    }
+
+    const { row, column } = adj[key];
+
+    if(row < 0 || column < 0 || row === maxRows || column === maxColumns ) {
+      delete adj[key];
```
Comment (faizanminhas) [inline 2020/11/index.js:73]: and you lost me!

File: 2020/11/index.js (hunk 1 above)
Comment (neverendingqs) [reply 2020/11/index.js:11]: Lol I had major struggles figuring how to skip [0, 0]. Didn't know the OR and NOT 0 check was the way to go.

File: 2020/11/index.js (hunk 3 above)
Comment (neverendingqs) [reply 2020/11/index.js:73]: Instead of checking for negative values, I just removed that direction from being considered in the next iteration of checks.

**Review by neverendingqs — COMMENTED** [review]
Thanks!

**Pull request merged**